src/arviz_plots/plots/evolutionplot.py

```diff
@@ -0,0 +1,449 @@
+    Returns
+    -------
+    PlotCollection
+    """
+    # initial defaults
+    if sample_dims is None:
+        sample_dims = rcParams["data.sample_dims"]
+    if isinstance(sample_dims, str):
+        sample_dims = [sample_dims]
+
+    # mutable inputs
+    if plot_kwargs is None:
+        plot_kwargs = {}
+    if pc_kwargs is None:
+        pc_kwargs = {}
+    else:
+        pc_kwargs = pc_kwargs.copy()
+
+    if stats_kwargs is None:
+        stats_kwargs = {}
+
+    # processing dt/group/coords/filtering
+    distribution = process_group_variables_coords(
+        dt, group=group, var_names=var_names, filter_vars=filter_vars, coords=coords
+    )
+
+    # set plot collection initialization defaults if it doesnt exist
+    if plot_collection is None:
+        if backend is None:
+            backend = rcParams["plot.backend"]
+        pc_kwargs.setdefault("col_wrap", 5)
+        pc_kwargs.setdefault(
+            "cols",
+            ["__variable__"]
+            + [dim for dim in distribution.dims if dim not in {"model"}.union(sample_dims)],
+        )
+        pc_kwargs["aes"] = pc_kwargs.get("aes", {}).copy()
+        if "chain" in distribution:
+            pc_kwargs["aes"].setdefault("overlay", ["chain"])  # so rug for each chain is overlaid
+            # doing this^ sets overlay: chain for each artist. But we only want overlay for the
+            # rug so .difference() has to be be applied to the aes_map defaults if not wanted
+        if "model" in distribution:
+            pc_kwargs["aes"].setdefault("color", ["model"])
+            # setting x aesthetic to np.linspace(-x_diff/3, x_diff/3, length of 'model' dim)
+            # x_diff = span of x axis (1) divided by number of points to be plotted (n_points)
+            x_diff = 1 / n_points
+            if "x" not in pc_kwargs:
+                pc_kwargs["x"] = np.linspace(-x_diff / 3, x_diff / 3, distribution.sizes["model"])
+            pc_kwargs["aes"].setdefault("x", ["model"])
+        aux_dim_list = [dim for dim in pc_kwargs["cols"] if dim != "__variable__"]  # for divergence
+        plot_collection = PlotCollection.wrap(
+            distribution,
+            backend=backend,
+            **pc_kwargs,
+        )
+    else:
+        aux_dim_list = list(
+            set(
+                dim for child in plot_collection.viz.children.values() for dim in child["plot"].dims
+            )
+        )
+
+    # set plot collection dependent defaults (like aesthetics mappings for each artist)
+    if aes_map is None:
+        aes_map = {}
+    else:
+        aes_map = aes_map.copy()
+    aes_map.setdefault("bulk", plot_collection.aes_set.difference({"overlay"}))
+    aes_map.setdefault("tail", plot_collection.aes_set.difference({"overlay"}))
+    aes_map.setdefault("mean", plot_collection.aes_set)
+    aes_map.setdefault("mean", ["linestyle", "-"])
+    aes_map.setdefault("sd", plot_collection.aes_set)
+    aes_map.setdefault("sd", ["linestyle", "-"])
```

Comment:
I think by default we don't want any aesthetic mapping to be applied to either mean nor sd, so I would remove these 4 lines. Also note the lines 2 and 4 are not doing anything because the lines right before already define that key, but if they did, it would be incorrect. Values in the `aes_map` dict should be sets of strings that represent aesthetics.

This seems a confusion with setting a default linestyle in `plot_kwargs`. If you want an aesthetic property to always have the same value, use `plot_kwargs` if instead you want it to depend on a dimension/variable use `aes` in `pc_kwargs` to define the mapping and `aes_map` to indicate to which visual elements should the mapping be applied too.

There is nothing about this concepts in https://arviz-plots.readthedocs.io/en/latest/contributing/new_plot.html because that doc is for writing plots and these are usage questions which are therefore out of scope. The usage docs are instead https://arviz-plots.readthedocs.io/en/latest/tutorials/plots_intro.html and to a much lesser extent https://arviz-plots.readthedocs.io/en/latest/tutorials/intro_to_plotcollection.html